src/openapi.ts

```diff
@@ -0,0 +1,651 @@
+      asyncComponents.schemas = convertSchemas(openapi.components.schemas);
+    }
+
+    if (openapi.components.securitySchemes) {
+      asyncComponents.securitySchemes = convertSecuritySchemes(openapi.components.securitySchemes);
+    }
+
+    if (openapi.components.parameters) {
+      asyncComponents.parameters = {};
+      for (const [name, param] of Object.entries(openapi.components.parameters)) {
+        if (!isRefObject(param)) {
+          asyncComponents.parameters[name] = convertParameter(param);
+        } else {
+          asyncComponents.parameters[name] = param; 
+        }
+      }
+    }
+
+    if (openapi.components.responses) {
+      asyncComponents.messages = convertComponentResponsesToMessages(openapi.components.responses);
+    }
+
+    if (openapi.components.requestBodies) {
+      asyncComponents.messageTraits = convertRequestBodiesToMessageTraits(openapi.components.requestBodies);
+    }
+
+    if (openapi.components.headers) {
+      asyncComponents.messageTraits = {
+        ...(asyncComponents.messageTraits || {}),
+        ...convertHeadersToMessageTraits(openapi.components.headers)
+      };
+    }
+
+    if (openapi.components.examples) {
+      asyncComponents.examples = openapi.components.examples;  
+    }
+  }
+
+  return removeEmptyObjects(asyncComponents);
+}
+
+/**
+ * Converts OpenAPI Schema Objects to AsyncAPI Schema Objects.
+ * @param {Record<string, any>} schemas - The OpenAPI Schema Objects to convert.
+ * @returns {Record<string, any>} The converted AsyncAPI Schema Objects.
+ */
+function convertSchemas(schemas: Record<string, any>): Record<string, any> {
+  const convertedSchemas: Record<string, any> = {};
+
+  for (const [name, schema] of Object.entries(schemas)) {
+    convertedSchemas[name] = {
+      schemaFormat: 'application/vnd.oai.openapi;version=3.0.0',
+      schema: schema
+    };
+  }
+
+  return convertedSchemas;
+}
+
+/**
+ * Converts a single OpenAPI Schema Object to an AsyncAPI Schema Object.
+ * @param {any} schema - The OpenAPI Schema Object to convert.
+ * @returns {any} The converted AsyncAPI Schema Object.
+ */
+function convertSchema(schema: any): any {
+  if (isRefObject(schema)) {
+    return schema;
+  }
+
+  const convertedSchema: any = { ...schema };
+
+  if (schema.properties) {
+    convertedSchema.properties = {};
+    for (const [propName, propSchema] of Object.entries(schema.properties)) {
+      convertedSchema.properties[propName] = convertSchema(propSchema);
+    }
+  }
+
+  if (schema.items) {
+    convertedSchema.items = convertSchema(schema.items);
+  }
+
+  ['allOf', 'anyOf', 'oneOf'].forEach(key => {
+    if (schema[key]) {
+      convertedSchema[key] = schema[key].map(convertSchema);
+    }
+  });
+
+  // Handle formats
+  if (schema.format === 'date-time') {
+    convertedSchema.format = 'date-time';
+  } else if (schema.format === 'byte' || schema.format === 'binary') {
+    delete convertedSchema.format;
+  }
+
+  return convertedSchema;
+}
```

Comment:
Is this needed anymore?

Reply:
Yes we dont need this anymore, updated it